include/llbuild/BuildSystem/BuildExecutionQueue.h

```diff
@@ -107,14 +107,22 @@ class BuildExecutionQueue {
   /// \param environment The environment to launch with. If empty, the process
   /// environment will be inherited.
   ///
+  /// \param hasPartialOutput Callback for stdout. Might be called multiple
+  /// times as output from output stream is available.
+  ///
+  /// \param hasPartialError Callback for stderr. Might be called multiple
+  /// times as output from error stream is available.
+  ///
   /// \returns True on success.
   //
   // FIXME: This interface will need to get more complicated, and provide the
   // command result and facilities for dealing with the output.
   virtual bool
   executeProcess(QueueJobContext* context,
                  ArrayRef<StringRef> commandLine,
-                 ArrayRef<std::pair<StringRef, StringRef>> environment) = 0;
+                 ArrayRef<std::pair<StringRef, StringRef>> environment,
+                 std::function<void(StringRef)> hasPartialOutput = nullptr,
+                 std::function<void(StringRef)> hasPartialError = nullptr) = 0;
```

Comment:
I think we should change the API behaviour to:
* Client must provide callbacks for stdout and stderr if they're interested in it.
* Remove `commandHasOutput` from delegate.